extensions/lifecycle/lifecycleProducer/LifecycleProducer.js

```diff
@@ -0,0 +1,256 @@
+'use strict'; // eslint-disable-line
+
+const async = require('async');
+const AWS = require('aws-sdk');
+const http = require('http');
+const Logger = require('werelogs').Logger;
+
+const BackbeatProducer = require('../../../lib/BackbeatProducer');
+const BackbeatConsumer = require('../../../lib/BackbeatConsumer');
+const AccountCredentials =
+    require('../../../lib/credentials/AccountCredentials');
+const safeJsonParse = require('../util/safeJsonParse');
+
+const CONSUMER_FETCH_MAX_BYTES = 5000020;
+
+class LifecycleProducer {
+    /**
+     * Constructor of LifecycleProducer
+     * @param {Object} zkConfig - zookeeper config
+     * @param {Object} lcConfig - lifecycle config
+     * @param {Object} s3Config - s3 config
+     * @param {String} s3Config.host - host ip
+     * @param {String} s3Config.port - port
+     * @param {Object} authConfig - auth config
+     * @param {String} authConfig.account - account name
```

Comment:
Technically, only one of either `account` or `vault` sections is mandatory, so you can specify both as optional.

Reply:
Updated.